Xamarin.Forms.Core/Shell/Shell.cs

```diff
@@ -324,6 +324,25 @@ void IShellController.UpdateCurrentState(ShellNavigationSource source)
 
 		public static Shell CurrentShell => Application.Current?.MainPage as Shell;
 
+		Uri GetAbsoluteUri(Uri relativeUri)
+		{
+			if (CurrentItem == null)
+				throw new InvalidOperationException("Relative path is used after selecting Current item.");
+
+			Element item = CurrentItem;
+			var list = new List<string> { relativeUri.OriginalString.Trim('/') };
+			while (item != null)
+			{
+				var route = Routing.GetRoute(item)?.Trim('/');
+				if (string.IsNullOrEmpty(route))
+					break;
+				list.Insert(0, route);
+				item = item.Parent;
+			}
+			var builder = new UriBuilder(RouteScheme) { Path = string.Join("/", list) };
```

Comment:
What about `Query` and `Fragment`? I think they should be preserved.

Reply:
I agree with this